### PR TITLE
Refactoring constants reduce lines

### PR DESCRIPTION
**Pull request by springwq**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR aims to reduce lines `constants.rb`, I tried to split it into two files.

https://codeclimate.com/github/thepracticaldev/dev.to/app/lib/constants.rb/source#issue-44d183a7091fe44e571161f871d2669e

## Related Tickets & Documents

https://github.com/forem/forem/issues/3739

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -592,7 +592,7 @@ def validate_mastodon_url
     return if mastodon_url.blank?
 
     uri = URI.parse(mastodon_url)
-    return if uri.host&.in?(Constants::ALLOWED_MASTODON_INSTANCES)
+    return if uri.host&.in?(Constants::AllowedMastodon::INSTANCES)
```
Comment (rhymes) [inline app/models/user.rb:595]: what about `Constants::Mastodon::ALLOWED_INSTANCES` ?

Comment (springwq) [reply app/models/user.rb:595]: Hi @rhymes , in order to reduce the lines of  file `contents.rb`, I tried to seperate  it into two files: `constants/allowed_mastodon.rb` and `constants/site_config.rb`.

So `Constants::Mastodon::ALLOWED_INSTANCES` is actually the same as `Constants::ALLOWED_MASTODON_INSTANCES`.

Comment (springwq) [reply app/models/user.rb:595]: Hi @rhymes , `Constants::Mastodon::ALLOWED_INSTANCES` looks good, I will revise my PR. 
Thank you very much for your code review!

Comment (rhymes) [reply app/models/user.rb]: Don't forget to mark the PR "ready for review" when it is :-)

Comment (springwq) [reply app/models/user.rb]: Thanks for reminding, I am waiting for CI finished. 😄

Comment (rhymes) [reply app/models/user.rb]: there's a typo in the name of the constants, see https://travis-ci.com/github/forem/forem/jobs/370710544#L1603

Comment (springwq) [reply app/models/user.rb]: @rhymes Fixed the typo, please help to review again. 😃

**Comment by rhymes** [comment]
Thanks @springwq

**Pull request merged**